File: src/Pages/Reservation.js
import { useEffect, useState } from "react";

function Reservation() {
  const [res, setRes] = useState([]);

  useEffect(() => {
    fetch("reservations")
      .then((res) => {
        return res.json();
      })
      .then(function (data) {
        setRes(data);
      });
  }, []);

  return (
    <div>
      <h2>예약 리스트</h2>
      <ul>
        {res.map((res) => (
          <li key={`%{res.id}`}>
            <p>ID: {res.id}</p>
            <p>First Name: {res.firstName}</p>
            <p>Last Name: {res.lastName}</p>
            <p>Email: {res.email}</p>
            <p>Phone: {res.phone}</p>
            <p>Room Number: {res.roomNum}</p>
            <p>Room Price: {res.roomPrice}</p>
            <p>Room Count: {res.roomCount}</p>
            <p>Room Role: {res.roomRole}</p>
          </li>
        ))}
      </ul>
    </div>
  );
}
export default Reservation;

File: src/main/java/com/teamprooject/team/reservation/Reservation.java
package com.teamprooject.team.reservation;

import com.teamprooject.team.room.RoomRole;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document(collection = "reservations")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Reservation {

    @Id
    private String id;

    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private String roomId;
    private RoomRole roomRole;
    private int roomPrice;
    private int roomCount;
    private LocalDate startDate;
    private LocalDate endDate;
}
